### ISSUE TITLE
Replace `upgrade` command with more generic `tool` command

### ISSUE THREAD

**Issue by freakboy3742** [beeware/briefcase#747]
The `briefcase upgrade` command provides a convenient interface to force an update of the tools Briefcase is managing. However, there are other "tool related" features that would be helpful to expose:

* Versions of currently managed tools
* Versions of tools that exist, but Briefcase can't update (e.g., Xcode)
* Versions of internal components (e.g., the SDKs managed by the Android SDKmanager
* The ability to *execute* tools (e.g., sdkmanager)

We currently compute the version of tools as part of the verification process, but we don't generally surface that information except in debug mode.

The Android backend will dump the contents of the sdkmanger during debug, but doesn't expose a "non-debug" way to list the same information.

The only way to easily invoke the Android SDK manager at present is to use `-v` to see the command line and environment used to issue commands, and manually reproduce those commands. 

There isn't an easy way to start a Docker shell in a project's environment.

**Describe the solution you'd like**

One option would be to replace the `briefcase upgrade` command with a new `briefcase tool` command. This new command would allow for the following usage patterns:

* `briefcase tool --list` - list all tools used by Briefcase, including the version numbers. 
* `briefcase tool --upgrade` - upgrade all tools currently managed by Briefcase
* `briefcase tool --upgrade <toolname>` - upgrade all tools currently managed by Briefcase
* `briefcase tool --run <toolname> <args...>` - run a specific tool with the provided arguments. This should configure any 

**Describe alternatives you've considered**

It may be preferable retain the `upgrade` command as-is, extend it to include *more* of the managed tools (e.g., it doesn't currently display the Xcode version because there's no upgrade path), and include the version numbers as part of the `upgrade --list` output.

The execution of specific tools would then be a feature of a specific backend - probably the `run` command. For example `briefcase run android --sdkmanager <args>` would invoke the SDK manager. This would then allow each backend to provide very specific tool executions, rather than needing to expose a generic API for gathering tools into the upgrade command.

**Comment by freakboy3742** [comment]
Addendum: the contents of `briefcase tools --list` should be included as part of the output when the user runs any command with `-v` - if we ask a user to provide debug detail, automatically getting a dump of all the tools and their versions as part of that dump would be very helpful.

**Comment by mhsmith** [comment]
`briefcase upgrade` should be deprecated, because every single new user probably confuses it with `briefcase update`.

**Comment by mhsmith** [comment]
A couple of cosmetic points about the command line format, arising from #2639:

* We've generally implemented "verbs" as subcommands rather than flags, which would suggest `briefcase tool run` rather than `briefcase tool --run`.
* Some "tools" may provide multiple commands, suggesting a format like `briefcase tool run android_sdk adb ARGS ...`

**Comment by freakboy3742** [comment]
Agreed on both these points. The only thing I'd add is that we may want to simplify `briefcase tool run android_sdk adb ...` into just `briefcase tool adb ...` for the sake of simplifying the end-user experience. 

That would mean there's potential collision between tool management commands (if we go down the `briefcase tool upgrade` path, instead of `briefcase tool adb --upgrade`), and between tools (e.g., if the Windows ecosystem gains a tool with the name "adb" for some reason) - but overall, I think the risk of both of these is fairly low, and is resolvable by using some sort of naming prefix (e.g., `windows-adb`) on *our* interface.
